### PR TITLE
feat!: `chainHub.getAsset` requires `srcChainName` parameter

### PR DESCRIPTION
**Pull request by 0xpatrickdev**
refs: #10445 

## Description
IBC Denoms are unique to a chain but not all not all chains. e.g., if `channel-0` points to `osmosis` for two chains, the `uosmo` denom  will be `ibc/ED07A3391A112B175915CD8FAF43A2DA8E4790EDE12566649D0C2F97716B8518` for both.

This change requires callers to specify the `holdingChainName` for the denom they wish to retrieve information about.

### Security Considerations
None

### Scaling Considerations
None

### Documentation Considerations
None

### Testing Considerations
Updates existing tests. Motivated from work in #10571 which surfaced this issue.

### Upgrade Considerations
Breaking change, but for library code that will be part on NPM or FUSDC release.

### CODE REVIEW

**Comment by 0xpatrickdev** [comment]
> ✘ [fail]: user can open a vault under debt limit The IST balance should increase by the minted amount

[This](https://github.com/Agoric/agoric-sdk/actions/runs/12091720759/job/33720468605?pr=10588#step:10:6230) seems unrelated to these changes and is likely a flake.

File: packages/orchestration/src/orchestration-api.ts [hunk 1]
```diff
@@ -146,6 +146,7 @@ export interface Orchestrator {
     IssuingChain extends keyof KnownChains,
   >(
     denom: Denom,
+    holdingChainName: HoldingChain,
```
Comment (dckc) [inline packages/orchestration/src/orchestration-api.ts]: +cc @dtribble @mitdralla

Comment (dckc) [reply packages/orchestration/src/orchestration-api.ts]: why is `holdingChainName` constrained to a static list of known chains?

OK, I guess I see why for now. Tolerable tech debt, I suppose.

In due course, it seems like `getDenomInfo` should be parameterized by...

```
  getDenomInfo: <
    HoldingChainInfo extends ChainInfo,
    IssuingChainInfo extends ChainInfo,
  >(
```

and maybe even the `ConnectionInfo` between them.

File: packages/orchestration/test/facade-durability.test.ts [hunk 2]
```diff
@@ -213,10 +213,14 @@ test('asset / denom info', async t => {
...
     // eslint-disable-next-line no-shadow
     async (orc, { brand }) => {
-      const c1 = await orc.getChain(mockChainInfo.chainId);
+      const c1 = await orc.getChain('mock');
 
       {
-        const actual = orc.getDenomInfo('utoken1');
+        const actual = orc.getDenomInfo(
+          'utoken1',
+          // @ts-expect-error 'mock' not a KnownChain
```
Comment (dckc) [inline packages/orchestration/test/facade-durability.test.ts:221]: doesn't seem like that should be an error

**Review by dckc — APPROVED** [review]
this seems about right

File: packages/orchestration/src/orchestration-api.ts (hunk 1 above)
Comment (0xpatrickdev) [reply packages/orchestration/src/orchestration-api.ts]: > why is holdingChainName constrained to a static list of known chains?

Something to potentially solve in #9992. In the (stale) PR that attempted to close it, allowing `string` for `chainName` was an approach discussed.

**Comment by turadg** [comment]
@mergify refresh

**Pull request merged**